R/trace_calls.R

```diff
@@ -131,7 +123,7 @@ new_counter <- function(src_ref, parent_functions) {
 #' @keywords internal
 count <- function(key) {
   .counters[[key]]$value <- .counters[[key]]$value + 1L
-  if (isTRUE(getOption("covr.record_tests"))) count_test(key)
+  if (isTRUE(.current_test$record)) count_test(key)
```

Comment:
When profiling `covr`, I found that calls to `getOption` was introducing non-negligible overhead to the "vanilla" `covr` path. Storing the option when the counters are reset and using the value more directly cuts this down quite a lot, speeding up vanilla `covr` by ~5% for this `rlang` stress test. 

Most packages won't hit `count` nearly as often as `rlang` does, but for packages that do, this is a meaningful performance gain.